pkg/types/vector.go

```diff
@@ -55,6 +56,28 @@ func InitVectorFloat32(dims int) VectorFloat32 {
 	return VectorFloat32{data: data}
 }
 
+// CheckVectorDimValid checks if the vector's dimension is valid.
+func CheckVectorDimValid(dim int) error {
+	const (
+		maxVectorDimension = 16000
```

Comment:
the maximum dimension supported in MySQL is 16383. We should align to it according to the feature spec. /cc @breezewish

Reply:
Nice catch. Please also add a TiDB-CSE PR to update this limit. @EricZequan

Reply:
Roger that.🫡